File: java/AutomationTools-Series/bitbucket-pipelines/src/main/java/localpage/CheckoutPage.java
/*
 * Copyright 2021 Automate The Planet Ltd.
 * Author: Teodor Nikolov
 * Licensed under the Apache License, Version 2.0 (the "License");
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package localpage;

import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;

import java.net.URL;

public class CheckoutPage {
    private final WebDriver driver;
    private final String url = "https://getbootstrap.com/docs/5.0/examples/checkout/";

    public CheckoutPage(WebDriver driver) {
        this.driver = driver;
    }

    protected CheckoutPageElements elements() {
        return new CheckoutPageElements(driver);
    }

    public CheckoutPageAssertions assertions() {
        return new CheckoutPageAssertions(driver);
    }

    public void navigate() {
        driver.navigate().to(url);
    }

    public void fillInfo(ClientInfo clientInfo) {
        elements().firstName().sendKeys(clientInfo.getFirstName());
        elements().lastName().sendKeys(clientInfo.getLastName());
        elements().username().sendKeys(clientInfo.getUsername());
        elements().email().sendKeys(clientInfo.getEmail());
        elements().address1().sendKeys(clientInfo.getAddress1());
        elements().address2().sendKeys(clientInfo.getAddress2());
        elements().country().selectByIndex(clientInfo.getCountry());
        elements().state().selectByIndex(clientInfo.getState());
        elements().zip().sendKeys(clientInfo.getZip());
        elements().cardName().sendKeys(clientInfo.getCardName());
        elements().cardNumber().sendKeys(clientInfo.getCardNumber());
        elements().cardExpiration().sendKeys(clientInfo.getCardExpiration());
        elements().cardCVV().sendKeys(clientInfo.getCardCVV());

        var js = (JavascriptExecutor)driver;
        js.executeScript("arguments[0].click();", elements().submitButton());
    }
}

File: dotnet/AutomationTools-Series/BitbucketPipelines/Pages/CheckoutPage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using OpenQA.Selenium;

namespace BitbucketPipelines.Pages
{
    public partial class CheckoutPage
    {
        private const string URL = "https://getbootstrap.com/docs/5.0/examples/checkout/";
        private IWebDriver _driver;

        public CheckoutPage(IWebDriver driver)
        {
            _driver = driver;
        }

        public void Navigate()
        {
            _driver.Navigate().GoToUrl(URL);
        }

        public void FillInfo(ClientInfo clientInfo)
        {
            FirstName.SendKeys(clientInfo.FirstName);
            LastName.SendKeys(clientInfo.LastName);
            Username.SendKeys(clientInfo.Username);
            Email.SendKeys(clientInfo.Email);
            Address1.SendKeys(clientInfo.Address1);
            Address2.SendKeys(clientInfo.Address2);
            Country.SelectByIndex(clientInfo.Country);
            State.SelectByIndex(clientInfo.State);
            Zip.SendKeys(clientInfo.Zip);
            CardName.SendKeys(clientInfo.CardName);
            CardNumber.SendKeys(clientInfo.CardNumber);
            CardExpiration.SendKeys(clientInfo.CardExpiration);
            CardCVV.SendKeys(clientInfo.CardCVV);
            ClickSubmitButton();
        }

        private void ClickSubmitButton()
        {
            ((IJavaScriptExecutor)_driver).ExecuteScript("arguments[0].click();", SubmitButton);
        }
    }
}
